kuka_rsi_hw_interface/src/rsi_command_handler.cpp

```diff
@@ -0,0 +1,325 @@
+        throw std::logic_error(err_buf_);
+      }
+      if (*buffer_it != '<') {
+        auto i = sprintf(err_buf_, "In \"");
+        std::snprintf(err_buf_ + i, node_name.length_ + 1, "%s", node_name.data_ptr_);
+        i += node_name.length_;
+        std::sprintf(err_buf_ + i, "\" param syntax error found");
+        throw std::logic_error(err_buf_);
+      }
+    } else {
+      // node base has childs
+      for (auto && child : element.GetChilds()) {
+        decodeNode(child, buffer, buffer_it, buffer_size);
+      }
+    }
+    for (; (size_t)(buffer_it - buffer) < buffer_size && *buffer_it != '<'; buffer_it++) {
+    }
+    if ((size_t)(buffer_it - buffer) >= buffer_size) {
+      throw std::range_error("Out of the buffers range");
+    }
+    buffer_it++;
+    if (*(buffer_it) != '/') {
+      std::sprintf(
+        err_buf_,
+        "Syntax error, while node end in the %s node",
+        element.GetName().c_str());
+      throw std::logic_error(err_buf_);
+    }
+    buffer_it++;
+    if (!element.IsNameValid(node_name, buffer_it)) {
+      auto i = std::sprintf(err_buf_, "The detected name (");
+      std::snprintf(err_buf_ + i, node_name.length_ + 1, "%s", node_name.data_ptr_);
+      i += node_name.length_;
+      std::sprintf(
+        err_buf_ + i, ") does not match the elements name: %s",
+        element.GetName().c_str());
+      throw std::logic_error(err_buf_);
+    }
+    if (*buffer_it != '>') {
+      std::sprintf(
+        err_buf_,
+        "Syntax error, while node end in the %s node",
+        element.GetName().c_str());
+      throw std::logic_error(err_buf_);
+    }
+  }
+}
+
+void RSICommandHandler::encodeNode(
+  xml::XMLElement & element, char * const buffer, char * & buffer_it,
+  const size_t buffer_size)
+{
+  auto idx = sprintf(buffer_it, "<%s ", element.GetName().c_str());
+  if ((size_t)(buffer_it - buffer) + idx >= buffer_size) {
+    throw std::range_error("Out of the buffers range");
+  }
+  buffer_it += idx;
+  bool isBaseLessNode = false;
+  if (element.GetChilds().size() <= 0) {
+    isBaseLessNode = true;
+  }
+  for (auto && param : element.GetParams()) {
+    if (element.GetName() == param.first) {
+      isBaseLessNode = false;
+    } else {
+      idx = sprintf(buffer_it, "%s=\"", param.first.c_str());
+      if ((size_t)(buffer_it - buffer) + idx >= buffer_size) {
+        throw std::range_error("Out of the buffers range");
+      }
+      buffer_it += idx;
+      param.second.ParamSprint(buffer_it, buffer, buffer_size);
+      idx = sprintf(buffer_it, "\" ");
+      if ((size_t)(buffer_it - buffer) + idx >= buffer_size) {
+        throw std::range_error("Out of the buffers range");
+      }
+      buffer_it += idx;
+    }
+  }
+  if (isBaseLessNode) {
+    idx = sprintf(buffer_it, "/>");
+    if ((size_t)(buffer_it - buffer) + idx >= buffer_size) {
+      throw std::range_error("Out of the buffers range");
+    }
+    buffer_it += idx;
+  } else {
+    buffer_it--;
```

Comment:
if you added a whitespace before printing a parameter name instead of after the element name, this would not be necessary IMO and would make the code a little cleaner

Reply:
modified so there is no buffer_it--